src/wp-admin/includes/class-wp-plugins-list-table.php

```diff
@@ -796,26 +800,60 @@ public function single_row( $item ) {
 			if ( $screen->in_admin( 'network' ) ) {
 				if ( $is_active ) {
 					if ( current_user_can( 'manage_network_plugins' ) ) {
-						$actions['deactivate'] = sprintf(
-							'<a href="%s" id="deactivate-%s" aria-label="%s">%s</a>',
-							wp_nonce_url( 'plugins.php?action=deactivate&amp;plugin=' . urlencode( $plugin_file ) . '&amp;plugin_status=' . $context . '&amp;paged=' . $page . '&amp;s=' . $s, 'deactivate-plugin_' . $plugin_file ),
-							esc_attr( $plugin_id_attr ),
-							/* translators: %s: Plugin name. */
-							esc_attr( sprintf( _x( 'Network Deactivate %s', 'plugin' ), $plugin_data['Name'] ) ),
-							__( 'Network Deactivate' )
-						);
+						if ( $has_active_dependents ) {
+							$dependent_names = WP_Plugin_Dependencies::get_dependent_names( $plugin_file );
+
+							$screen_reader_text = sprintf(
+								__( 'This plugin cannot be deactivated as %s depends on it.' ),
+								esc_html( array_shift( $dependent_names ) )
```

Comment:
The `array_shift()` may produce a dependent that is already deactivated, making it unclear which remaining dependents need to be deactivated. It would be ideal if we could instead show a plugin that is active.

Should we consider extending the following public API methods with an additional `string|array $status = ''` parameter?
- `::get_dependent_names( $plugin_file, $status = '' )`
- `::get_dependency_names( $plugin_file, $status = '' )`
- `::get_dependents( $slug, $status = '' )`
- `::get_dependencies( $plugin_file, $status = '' )`

If `$status` is not empty, we can then filter the dependents/dependencies down to those that are active, inactive, installed, uninstalled, or any mixture of these.

Something like:
```php
/*
 * $status is now array( 'active' => 0, 'inactive' => 1, 'installed' => 2, 'uninstalled' => 3 )
 *
 * This avoids doing in_array() checks for each plugin, instead opting for the more performant isset().
 * While both array_flip() and in_array() are O(n), array_flip() will only run once.
 */
$status = array_flip( (array) $status );
foreach ( self::get_dependents( $slug ) as $dependent ) {
    if (
        empty( $status ) ||
        ( isset( $status['active'] ) && is_plugin_active( $dependent ) ) ||
        ( isset( $status['inactive'] ) === $status && ! is_plugin_active( $dependent ) ) ||
        ( isset( $status['installed'] ) === $status && isset( self::$plugins[ $dependent ] ) ) ||
        ( isset( $status['uninstalled'] ) === $status && ! isset( self::$plugins[ $dependent ] ) )
    ) {
        // The existing code.
    }
}
```

Your call to `::get_dependent_names()` would then be:
```php
$dependent_names = WP_Plugin_Dependencies::get_dependent_names( $plugin_file, 'active' );
```
and you could continue using `esc_html( array_shift( $dependent_names ) )`.

Reply:
Good catch :) Was thinking about doing this too, even started coding it. The (more important?) question here is: would it be more useful/less confusing if these warnings/messages are so targeted? Or perhaps can be more general so the `$screen_reader_text matches more closely the warnings that are shown on the screen? 

Perhaps better to make these more like the other warnings for now. Can always change them later, and add that param to be able to get only active plugins.